plugins/binanceExchange_ws.go

```diff
@@ -180,6 +262,98 @@ func subcribeTicker(symbol string, state *mapEvents) (*stream, error) {
 
 }
 
+func subcribeUserStream(listenKey string, state *mapEvents) (*stream, error) {
+
+	userStreamLock := sync.Mutex{}
+
+	wsUserStreamExecutinReportHandler := func(message []byte) {
+
+		event := &eventExecutionReport{}
+		err := json.Unmarshal(message, event)
+
+		if err != nil {
+			log.Printf("Error unmarshal %s to eventExecutionReport\n", string(message))
+			return
+		}
+
+		userStreamLock.Lock()
```

Comment:
why do we need to lock here?
wondering in which scenario may this be executed twice, or maybe I've missed something

Reply:
this is to avoid any misusage of the callback and to not have invalid data, because I am modifying the slice in the callback and the possibility for that callback to be executed by two goroutines exists I've added the mutex

Reply:
Ok will look at this a little more closely when I’m on a computer